engine/src/main/scala/cromwell/services/KeyValueServiceActor.scala

```diff
@@ -0,0 +1,69 @@
+package cromwell.services
+
+import akka.actor.{Actor, ActorRef, Props}
+import com.typesafe.config.Config
+import cromwell.backend.BackendJobDescriptor
+import cromwell.engine.db.DataAccess
+import cromwell.services.KeyValueServiceActor._
+
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+
+object KeyValueServiceActor {
+  trait KvMessage
+  trait KvAction extends KvMessage
+  trait KvResponse extends KvMessage
+
+  case class KvPut(jobDescriptor: BackendJobDescriptor, key: String, value: String) extends KvAction
+  case class KvGet(jobDescriptor: BackendJobDescriptor, key: String) extends KvAction
+
+  case class KvPair(jobDescriptor: BackendJobDescriptor, key: String, value: Option[String]) extends KvResponse
```

Comment:
We currently allow `NULL` in the database for the corresponding column.  I forgot to make `KvPut` take an `value: Option[String]`